drivers/overlay/idmapped_utils.go

```diff
@@ -0,0 +1,181 @@
+//go:build linux
+// +build linux
+
+package overlay
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"syscall"
+	"unsafe"
+
+	"github.com/containers/storage/pkg/idtools"
+	"github.com/pkg/errors"
+	"golang.org/x/sys/unix"
+)
+
+type attr struct {
+	attrSet     uint64
+	attrClr     uint64
+	propagation uint64
+	userNs      uint64
+}
+
+const (
+	// MOUNT_ATTR_RDONLY - Mount read-only
+	MOUNT_ATTR_RDONLY = 0x00000001 //nolint:golint
+	// MOUNT_ATTR_NOSUID - Ignore suid and sgid bits
+	MOUNT_ATTR_NOSUID = 0x00000002 //nolint:golint
+	// MOUNT_ATTR_NODEV - Disallow access to device special files
+	MOUNT_ATTR_NODEV = 0x00000004 //nolint:golint
+	// MOUNT_ATTR_NOEXEC - Disallow program execution
+	MOUNT_ATTR_NOEXEC = 0x00000008 //nolint:golint
+	// MOUNT_ATTR__ATIME - Setting on how atime should be updated
+	MOUNT_ATTR__ATIME = 0x00000070 //nolint:golint
+	// MOUNT_ATTR_RELATIME - Update atime relative to mtime/ctime
+	MOUNT_ATTR_RELATIME = 0x00000000 //nolint:golint
```

Comment:
Quick question is this one correct?  No bits set?

Reply:
yes, it is correct:

```
$ grep MOUNT_ATTR_RELATIME /usr/include/linux/mount.h
#define MOUNT_ATTR_RELATIME	0x00000000 /* - Update atime relative to mtime/ctime. */
```

I am not sure if we should add all of these (added for completeness) and if they should be public.

Reply:
Can we use them from here?
 grep -r MOUNT_ATTR_NODEV .
./vendor/golang.org/x/sys/unix/zerrors_linux.go:	MOUNT_ATTR_NODEV                            = 0x4

Reply:
I've dropped the ones not used by the module and made the other 3 private